src/commands/step-functions/instrument.ts

```diff
@@ -0,0 +1,202 @@
+import {CloudWatchLogs, StepFunctions} from 'aws-sdk'
+import {Command} from 'clipanion'
+
+import {
+  createLogGroup,
+  enableStepFunctionLogs,
+  getStepFunction,
+  listStepFunctionTags,
+  putSubscriptionFilter,
+  tagLogGroup,
+  tagStepFunction,
+} from './aws'
+import {displayChanges, applyChanges} from './changes'
+import {
+  buildArn,
+  buildLogGroupName,
+  buildSubscriptionFilterName,
+  getStepFunctionLogGroupArn,
+  isValidArn,
+  parseArn,
+} from './helpers'
+import {
+  CreateLogGroupRequest,
+  PutSubscriptionFilterRequest,
+  TagStepFunctionRequest,
+  UpdateStepFunctionRequest,
+} from './interfaces'
+
+export class InstrumentStepFunctionsCommand extends Command {
+  public static usage = Command.Usage({
+    description: 'Subscribe Step Function Log Groups to a Datadog Forwarder',
+    examples: [
+      [
+        'View and apply changes to subscribe a Step Function Log Group to a Datadog Forwarder',
+        'datadog-ci step-functions instrument --step-function-arn arn:aws:states:us-east-1:000000000000:stateMachine:ExampleStepFunction --forwarder-arn arn:aws:lambda:us-east-1:000000000000:function:ExampleDatadogForwarder --env dev --service example-service',
+      ],
+      [
+        'View changes to subscribe a Step Function Log Group to a Datadog Forwarder',
+        'datadog-ci step-functions instrument --step-function-arn arn:aws:states:us-east-1:000000000000:stateMachine:ExampleStepFunction --forwarder-arn arn:aws:lambda:us-east-1:000000000000:function:ExampleDatadogForwarder --env dev --service example-service --dry-run',
+      ],
+      [
+        'View and apply changes to subscribe multiple Step Function Log Groups to a Datadog Forwarder',
+        'datadog-ci step-functions instrument --step-function-arn arn:aws:states:us-east-1:000000000000:stateMachine:ExampleStepFunction1 --step-function-arn arn:aws:states:us-east-1:000000000000:stateMachine:ExampleStepFunction2 --forwarder-arn arn:aws:lambda:us-east-1:000000000000:function:ExampleDatadogForwarder --env dev --service example-service',
+      ],
+    ],
+  })
+
+  private dryRun = false
+  private environment?: string
+  private forwarderArn?: string
```

Comment:
If its required, why set it as nullable?
```suggestion
  private forwarderArn: string
```

Reply:
I did it to avoid the below error. It looks like `private forwarderArn!: string` also works and is more appropriate for typing.

`Property 'forwarderArn' has no initializer and is not definitely assigned in the constructor.ts(2564)`